### ISSUE TITLE
Uncaught TypeError: Cannot read property 'split' of undefined

### ISSUE THREAD

**Issue by bronson** [hmatsuda/rails-transporter#84]
[Enter steps to reproduce below:]

1. Open a Helper
2. Use Transporter to open the corresponding Controller or Model.

Crashes every time.

Switching between model and spec, or controller and views, works just fine.

**Atom Version**: 1.3.3
**System**: Mac OS X 10.11.2
**Thrown From**: [rails-transporter](https://github.com/hmatsuda/rails-transporter) package, v1.6.0

### Stack Trace

Uncaught TypeError: Cannot read property 'split' of undefined

```
At /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:803

TypeError: Cannot read property 'split' of undefined
    at Function.module.exports.Range.fromText (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/range.js:47:19)
    at TextBuffer.module.exports.TextBuffer.setTextInRange (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:562:24)
    at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:550:26
    at TextBuffer.module.exports.TextBuffer.transact (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:798:18)
    at TextBuffer.module.exports.TextBuffer.setTextInRange (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:548:21)
    at TextBuffer.module.exports.TextBuffer.setText (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:488:19)
    at TextEditor.module.exports.TextEditor.setText (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/text-editor.js:752:26)
    at TextEditorView.module.exports.TextEditorView.setText (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-space-pen-views/lib/text-editor-view.js:56:25)
    at DialogView.module.exports.DialogView.setTargetFile (/Users/bronson/.atom/packages/rails-transporter/lib/dialog-view.coffee:43:17)
    at DialogView.setTargetFile (/Users/bronson/.atom/packages/rails-transporter/lib/dialog-view.coffee:1:1)
    at FileOpener.module.exports.FileOpener.openDialog (/Users/bronson/.atom/packages/rails-transporter/lib/file-opener.coffee:296:17)
    at FileOpener.module.exports.FileOpener.openController (/Users/bronson/.atom/packages/rails-transporter/lib/file-opener.coffee:76:8)
    at atom-workspace.atom.commands.add.rails-transporter:open-controller (/Users/bronson/.atom/packages/rails-transporter/lib/rails-transporter.coffee:42:29)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at CommandPaletteView.module.exports.CommandPaletteView.confirmed (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/command-palette/lib/command-palette-view.js:183:32)
    at CommandPaletteView.module.exports.SelectListView.confirmSelection (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-space-pen-views/lib/select-list-view.js:338:21)
    at space-pen-div.atom.commands.add.core:confirm (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-space-pen-views/lib/select-list-view.js:109:19)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:536:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:359:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:97:36)
    at HTMLDocument.<anonymous> (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
  2x -0:20.7.0 vim-mode-plus:move-down (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:12.9.0 command-palette:toggle (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:12.7.0 blur (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:10.2.0 editor:consolidate-selections (atom-text-editor.editor.mini.is-focused)
     -0:10.2.0 core:cancel (atom-text-editor.editor.mini.is-focused)
     -0:10.2.0 blur (atom-text-editor.editor.mini.is-focused)
     -0:09.7.0 fuzzy-finder:toggle-file-finder (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:09.6.0 blur (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:06.9.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:05.7.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  4x -0:05.7.0 blur (atom-text-editor.editor.mini.is-focused)
     -0:02.5.0 command-palette:toggle (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:02.3.0 blur (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:01.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:01.0 blur (atom-text-editor.editor.mini.is-focused)
     -0:01.0 rails-transporter:open-controller (atom-text-editor.editor.vim-mode-plus.normal-mode)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "metrics",
      "exception-reporting",
      "welcome",
      "background-tips"
    ],
    "themes": [
      "one-dark-ui",
      "oceanic-reef-syntax"
    ],
    "audioBeep": false,
    "packagesWithKeymapsDisabled": [
      "rspec",
      "ruby-test-switcher",
      "rails-transporter"
    ]
  },
  "rails-transporter": {}
}
```

### Installed Packages

```coffee
# User
cursor-history, v0.5.9
file-icons, v1.6.13
linter, v1.11.3
linter-eslint, v5.2.7
linter-haml, v1.1.0
linter-ruby, v1.2.1
linter-tidy, v1.0.1
rails-transporter, v1.6.0
react, v0.12.10
ruby-test-switcher, v1.2.0
vim-mode-plus, v0.14.1
white-cursor, v0.6.0

# Dev
No dev packages
```

**Comment by hmatsuda** [comment]
Thanks for reporting. v-1.6.1 fixes this issue.

**Comment by bronson** [comment]
I can still trigger this on 1.6.2.  To reproduce: open a controller spec, then ask Rails Transporter to open the Spec.  (yes, opening a spec while you're in a spec is a user mistake...  but I wouldn't expect it to crash)

**Atom Version**: 1.3.3
**System**: Mac OS X 10.11.2
**Thrown From**: [rails-transporter](https://github.com/hmatsuda/rails-transporter) package, v1.6.2


### Stack Trace

Uncaught TypeError: Cannot read property 'split' of undefined

```
At /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:803

TypeError: Cannot read property 'split' of undefined
    at Function.module.exports.Range.fromText (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/range.js:47:19)
    at TextBuffer.module.exports.TextBuffer.setTextInRange (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:562:24)
    at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:550:26
    at TextBuffer.module.exports.TextBuffer.transact (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:798:18)
    at TextBuffer.module.exports.TextBuffer.setTextInRange (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:548:21)
    at TextBuffer.module.exports.TextBuffer.setText (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:488:19)
    at TextEditor.module.exports.TextEditor.setText (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/text-editor.js:752:26)
    at TextEditorView.module.exports.TextEditorView.setText (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-space-pen-views/lib/text-editor-view.js:56:25)
    at DialogView.module.exports.DialogView.setTargetFile (/Users/bronson/.atom/packages/rails-transporter/lib/dialog-view.coffee:43:17)
    at DialogView.setTargetFile (/Users/bronson/.atom/packages/rails-transporter/lib/dialog-view.coffee:1:1)
    at FileOpener.module.exports.FileOpener.openDialog (/Users/bronson/.atom/packages/rails-transporter/lib/file-opener.coffee:323:17)
    at FileOpener.module.exports.FileOpener.openSpec (/Users/bronson/.atom/packages/rails-transporter/lib/file-opener.coffee:205:8)
    at atom-workspace.atom.commands.add.rails-transporter:open-spec (/Users/bronson/.atom/packages/rails-transporter/lib/rails-transporter.coffee:34:29)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at CommandRegistry.handleCommandEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:160:19)
    at dispatch (/Users/bronson/.atom/init.coffee:20:19)
    at atom-workspace.atom.commands.add.ex-command:spec (/Users/bronson/.atom/init.coffee:47:25)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at CommandPaletteView.module.exports.CommandPaletteView.confirmed (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/command-palette/lib/command-palette-view.js:183:32)
    at CommandPaletteView.module.exports.SelectListView.confirmSelection (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-space-pen-views/lib/select-list-view.js:338:21)
    at space-pen-div.atom.commands.add.core:confirm (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-space-pen-views/lib/select-list-view.js:109:19)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:536:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:359:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:97:36)
    at HTMLDocument.<anonymous> (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
  3x -0:15.6.0 blur (atom-text-editor.editor.vim-mode-plus.normal-mode)
     -0:11.3.0 command-palette:toggle (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:11.2.0 blur (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:09.1.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:09.1.0 blur (atom-text-editor.editor.mini.is-focused)
     -0:09.1.0 ex-command:spec (atom-text-editor.editor.vim-mode-plus.normal-mode)
     -0:09.1.0 rails-transporter:open-spec (atom-text-editor.editor.vim-mode-plus.normal-mode)
     -0:07.9.0 vim-mode-plus:reset-normal-mode (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
  3x -0:05.8.0 blur (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:04.4.0 vim-mode-plus:move-to-first-character-of-line-down (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:02.7.0 command-palette:toggle (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:02.6.0 blur (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:01.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:01.0 blur (atom-text-editor.editor.mini.is-focused)
     -0:01.0 ex-command:spec (atom-text-editor.editor.vim-mode-plus.normal-mode)
     -0:01.0 rails-transporter:open-spec (atom-text-editor.editor.vim-mode-plus.normal-mode)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "metrics",
      "exception-reporting",
      "welcome",
      "background-tips"
    ],
    "themes": [
      "one-dark-ui",
      "oceanic-reef-syntax"
    ],
    "audioBeep": false,
    "packagesWithKeymapsDisabled": [
      "rspec",
      "ruby-test-switcher",
      "rails-transporter"
    ]
  },
  "rails-transporter": {
    "viewFileExtensionFallbacks": [
      "html.slim",
      "html.haml"
    ]
  }
}
```

### Installed Packages

```coffee
# User
advanced-open-file, v0.14.1
atom-ternjs, v0.12.2
color-picker, v2.0.14
cursor-history, v0.5.9
file-icons, v1.6.13
highlight-selected, v0.11.1
language-haml, v0.23.14
language-viml, v0.3.0
linter, v1.11.3
linter-coffeelint, v1.1.2
linter-csslint, v1.2.0
linter-erb, v0.1.1
linter-eslint, v5.2.7
linter-haml, v1.1.0
linter-jscs, v3.2.2
linter-ruby, v1.2.1
linter-tidy, v1.0.1
oceanic-reef-syntax, v0.6.0
rails-transporter, v1.6.2
react, v0.12.10
rspec, v0.3.2
ruby-test-switcher, v1.2.0
vim-mode-plus, v0.15.0
white-cursor, v0.6.0

# Dev
vim-mode-plus, v0.11.0
```

**Comment by hmatsuda** [comment]
Sorry I fixed it by v1.6.3.

**Comment by bronson** [comment]
Works great!  Thanks.
